js/src/chat/chat-external-link.ts

```diff
@@ -0,0 +1,177 @@
+      <div class="modal position-relative d-block fade show">
+        <div class="modal-content">
+          <div class="modal-header">
+            <h5 class="modal-title">External Link</h5>
+            <button class="btn-close shinychat-btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+          </div>
+          <div class="modal-body">
+            <p>This link will take you to an external website:</p>
+            <p class="link-url text-break"></p>
+          </div>
+          <div class="modal-footer flex-wrap-reverse">
+            <button class="btn btn-sm btn-link shinychat-btn-always me-auto">Always open external links</button>
+            <div class="d-flex gap-2 justify-content-end">
+              <button autofocus class="btn btn-sm btn-primary shinychat-btn-proceed">Open Link</button>
+              <button class="btn btn-sm btn-outline-danger shinychat-btn-cancel">Cancel</button>
+            </div>
+          </div>
+          </div>
+        </div>
+      </div>
+    `
+
+    document.body.appendChild(this.dialog)
+
+    // Add event listeners to the buttons
+    const getBtn = (selector: string) =>
+      this.dialog?.querySelector(selector) as HTMLButtonElement
+
+    const btns = {
+      close: getBtn(".shinychat-btn-close"),
+      cancel: getBtn(".shinychat-btn-cancel"),
+      proceed: getBtn(".shinychat-btn-proceed"),
+      always: getBtn(".shinychat-btn-always"),
+    }
+
+    btns.close.addEventListener("click", () => this.handleCancel())
+    btns.cancel.addEventListener("click", () => this.handleCancel())
+    btns.proceed.addEventListener("click", () => this.handleProceed())
+    btns.always.addEventListener("click", () => this.handleAlways())
+
+    // Close the dialog when clicked outside (native dialog behavior)
+    this.dialog.addEventListener("click", (e) => {
+      if (e.target === this.dialog) {
+        this.handleCancel()
+        e.preventDefault()
+      }
+    })
+  }
+
+  private handleCancel(): void {
+    this.dialog?.close()
+    if (this.resolvePromise) {
+      this.resolvePromise(false)
+      this.resolvePromise = null
+    }
+  }
+
+  private handleProceed(): void {
+    this.dialog?.close()
+    if (this.resolvePromise) {
+      this.resolvePromise(true)
+      this.resolvePromise = null
+    }
+  }
+
+  private handleAlways(): void {
+    window.shinychat_always_open_external_links = true
+    this.handleProceed()
+  }
+}
+
+// Define the custom element
+if (!customElements.get("shinychat-external-link-dialog")) {
+  customElements.define(
+    "shinychat-external-link-dialog",
+    ChatExternalLinkDialog,
+  )
+}
+
+/**
+ * Shows a confirmation dialog for external links
+ * @param url The URL to confirm
+ * @returns A promise that resolves to true if confirmed, false otherwise
+ */
+export function showExternalLinkConfirmation(url: string): Promise<boolean> {
+  // Check if the browser supports HTMLDialogElement
+  if (typeof window.HTMLDialogElement !== "undefined") {
```

Comment:
Hmm, I'm not sure if this is needed? It'd be a surprise to me if Bootstrap depended on `<dialog>`, and also the browser support is pretty good https://caniuse.com/?search=HTMLDialog

Reply:
94% is good but also it's easy enough to check for support that I think it's worth keeping this.